src/main/java/Ui.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Scanner;
+
+public class Ui {
+    Scanner scanner;
+    private boolean isExit;
+    private Parser parser;
+
+    public Ui() {
+        scanner = new Scanner(System.in);
+        isExit = false;
+        parser = new Parser();
+    }
+
+    public void showLine() {
+        System.out.println("    ____________________________________________________________");
+    }
+
+    public void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        showLine();
+        System.out.println("     Hello! I'm Duke\n     What can I do for you?");
+        showLine();
+        System.out.println();
+    }
+
+    public void execute(TaskList tasks) {
+        String command = scanner.nextLine();
+        showLine();
+        if (command.equals("bye")) {
+            isExit = true;
+        } else if (command.equals("list")) {
+            System.out.println("     Here are the tasks in your list:");
+            for (int i = 0; i < tasks.getTasks().size(); i++) {
+                System.out.println(String.format("     %d.%s", i + 1, tasks.getTasks().get(i).toString()));
+            }
+            showLine();
+            System.out.println();
+        } else {
+            String[] commands = parser.parse(command);
+            if (commands[0].equals("delete")) {
+                System.out.println("     Noted. I've removed this task:");
+                int index = Integer.valueOf(commands[1]) - 1;
+                System.out.println("       " + tasks.getTasks().get(index).toString());
+                tasks.delete(index);
+                System.out.println(String.format("     Now you have %d tasks in the list.", tasks.getTasks().size()));
```

Comment:
Since you are reusing tasks.getTasks() multiple times, consider creating a variable just to store the reference